src/statsmodels/formula.jl

```diff
@@ -437,21 +441,21 @@ If there is an intercept in the model, that column occurs first and its
 Mixed-effects models include "random-effects" terms which are ignored when
 creating the model matrix.
 """
-function ModelMatrix(mf::ModelFrame)
+function ModelMatrix(T::Union{map(t->Type{t}, modelmatrixcontainertypes)...}, mf::ModelFrame)
     dfrm = mf.df
     terms = droprandomeffects(dropresponse!(mf.terms))
 
-    blocks = Matrix{Float64}[]
+    blocks = T[]
     assign = Int[]
     if terms.intercept
-        push!(blocks, ones(size(dfrm, 1), 1))  # columns of 1's is first block
-        push!(assign, 0)                        # this block corresponds to term zero
+        push!(blocks, convert(T, ones(size(dfrm, 1), 1)))  # columns of 1's is first block
```

Comment:
Since `blocks` is already typed, conversion should happen automatically without changing anything AFAIK.